react/ProductList.tsx

```diff
@@ -47,13 +50,56 @@ const ItemContextWrapper = memo<ItemWrapperProps>(function ItemContextWrapper({
   return <ItemContextProvider value={context}>{children}</ItemContextProvider>
 })
 
+const ProductGroup: StorefrontFunctionComponent<Props> = ({
+  items,
+  loading,
+  onQuantityChange,
+  onRemove,
+  renderOnView,
+  children,
+}) => {
+  const { hasBeenViewed, dummyElement } = useRenderOnView({
+    lazyRender: true,
+    offset: 900,
+  })
+
+  if (renderOnView && (!hasBeenViewed || items.length === 0)) {
+    return dummyElement
+  }
+
+  return (
+    <>
+      {items.map((item: Item) => (
+        <ItemContextWrapper
+          key={item.uniqueId + item.sellingPrice}
+          item={item}
+          loading={loading}
+          onQuantityChange={onQuantityChange}
+          onRemove={onRemove}
+        >
+          {children}
+        </ItemContextWrapper>
+      ))}
+    </>
+  )
+}
+
 const ProductList: StorefrontFunctionComponent<Props> = ({
   items,
   loading,
   onQuantityChange,
   onRemove,
+  renderOnView = true,
   children,
 }) => {
+  const props = {
```

Comment:
why recreate the props object here?

Reply:
👍

Reply:
I just wanted to group them in one variable to spread them all below in the `ProductGroup` instances in just one line.

At first, I was getting the `props` param without destructuring it and that wasn't needed. But since I started to need to define a default value for `renderOnView` I had to reconstruct the object this way to be less verbose below. I could have used the rest operator in the params after defining the default value for `renderOnView` and reconstruct it just spreading like `{ renderOnView, ...props}`, but I only see it now.

Maybe I'm missing a simpler way to do it... But I think from the moment I have to destruct in the param, next I'll have to reconstruct the `props` object, to be less verbose in the `ProductGroup` instances, right?

Reply:
personally I'd rather have the destructuring inside the function body and pass the props that you need the default value for explicitly to the other components, like

```jsx
const ProductList = (props) => {
  const { renderOnView = true } = props

  return (
    <MyChildComponent {...props} renderOnView={renderOnView} />
  )
}
```

Reply:
So much better. Thanks! It really didn't cross my mind.